lib/rules/arrow-body-style.js

```diff
@@ -174,15 +194,28 @@ module.exports = {
                                 );
                             }
 
+                            let n = node;
+
+                            if (isInOp) {
+                                while (n.type !== "ForStatement" && n.parent && (n.parent.type !== "BlockStatement" || n.parent.type !== "Program")) {
+                                    n = n.parent;
```

Comment:
It would be nice to extract this to a function.

Maybe `IsInForLoopInitializer(node)`, returns `true` or `false`.